beacon-chain/db/state.go

```diff
@@ -44,169 +36,98 @@ func (db *BeaconDB) InitializeState(genesisValidators []*pb.ValidatorRecord) err
 			return fmt.Errorf("failed to record block height: %v", err)
 		}
 
-		if err := mainChain.Put(zeroBinary, blockhash[:]); err != nil {
+		if err := mainChain.Put(zeroBinary, blockHash[:]); err != nil {
 			return fmt.Errorf("failed to record block hash: %v", err)
 		}
 
-		if err := blockBkt.Put(blockhash[:], blockEnc); err != nil {
-			return err
-		}
-
-		if err := chainInfo.Put(aStateLookupKey, aStateEnc); err != nil {
+		if err := blockBkt.Put(blockHash[:], blockEnc); err != nil {
 			return err
 		}
 
-		if err := chainInfo.Put(cStateLookupKey, cStateEnc); err != nil {
+		if err := chainInfo.Put(stateLookupKey, stateEnc); err != nil {
 			return err
 		}
 
 		return nil
 	})
 }
 
-// GetActiveState fetches the current active state.
-func (db *BeaconDB) GetActiveState() (*types.ActiveState, error) {
-	var aState *types.ActiveState
-	err := db.view(func(tx *bolt.Tx) error {
-		chainInfo := tx.Bucket(chainInfoBucket)
-
-		enc := chainInfo.Get(aStateLookupKey)
-		if enc == nil {
-			return nil
-		}
-
-		var err error
-		aState, err = createActiveState(enc)
-		return err
-	})
-
-	return aState, err
-}
-
-// GetCrystallizedState fetches the current active state.
-func (db *BeaconDB) GetCrystallizedState() (*types.CrystallizedState, error) {
-	var cState *types.CrystallizedState
+// GetState --
+func (db *BeaconDB) GetState() (*types.BeaconState, error) {
```

Comment:
We want to eventually have this accept a hash as a parameter, so that previous states can be retrieved as well, correct?

Reply:
Wouldnt this be just referring to the current state ? If we wanted to retrieve other previous states, we would need it to be in a separate bucket.